sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -881,7 +862,8 @@ object StateStore extends Logging {
             case NonFatal(e) =>
               logWarning(log"Error managing ${MDC(LogKeys.STATE_STORE_PROVIDER, provider)}, " +
                 log"stopping management thread", e)
-              threadPoolException.set(e)
```

Comment:
If the maint task for a given partition is failing all the time with the same exception - we don't have a way to inform the query right ? we kind of also don't have a mechanism to do backpressure to ensure that maintenance is caught up to the task execution

Reply:
Nope, there's no mechanism for that. But pulling on that thread, let's say that the maint task does fall behind and can never make progress. Then, the user will see that their checkpoint location never gets cleaned up. And, if they fail, due to not having a snapshot (in the case of changelog checkpointing), they will have slower recovery times. Is that correct?

Reply:
Yes correct